### ISSUE TITLE
Relevant fields need language attribute values

### ISSUE THREAD

**Issue by sdellis** [pulibrary/orangelight#1129]
Key fields would be Titles, Publisher, Series, Notes.
Roughly half our entire catalog (i.e., around 6 million non-english book titles) is inaccessible to screen reader users if we don't use lang attributes.

Related to: https://github.com/pulibrary/requests/issues/349

Screen readers need to know the language of text to read it correctly. If we know the language of a record, we should assign that language to the record title. https://www.w3.org/WAI/WCAG21/quickref/#language-of-parts

**Comment by sdellis** [comment]
Marc Fields Transcribed:
245 a,b,c
260 (264?)
880 (has language indicators?)

**Comment by tampakis** [comment]
I looked at a few of our French records and I agree that adding the language attribute fields based on the language field stored in solr. We currently have `language_facet` or `language_code_s` fields (although looking at pulibrary/requests#349 i guess we are looking at potentially indexing an addition code format). 

My only advice would be to only do at this language attribute if there is only one language associated with the work. We have a decent amount of multi-lingual works that are often still described fully in English: ie https://pulsearch.princeton.edu/catalog/SCSB-4622745

**Comment by mzelesky** [comment]
500 fields (note fields) should always be mostly in English as well, even if the language of the item described is not.

**Comment by sdellis** [comment]
@escowles where is this at? Can we move forward on a solution?

**Comment by christinach** [comment]
Used cld in this working branch https://github.com/pulibrary/orangelight/tree/1129-language-attr for language detection. This gem has license https://github.com/jtoy/cld `Licensed the same as Chrome`. From what I've seen chrome has a GBL license. Is this a problem for us to use it? 
In an example with italian it detected spanish. From the gems I've seen is the one that has the most recent updates. 
I will look into testing for the same examples another gem https://github.com/detectlanguage/detectlanguage-ruby that uses https://detectlanguage.com/

**Comment by christinach** [comment]
@escowles pointed out https://github.com/jtoy/cld/blob/master/LICENSE and said it seems to be ok.

**Comment by sdellis** [comment]
@christinach we did an experiment a few years ago using Google Translate APIs to detect languages in Finding Aid components. The CLD gem you are using may be a better approach for this, but we found it to be fairly easy to send a language detection request to GTranslate and get back an accurate response. The longer the text string, the more accurate the detection. See the curl and bash scripts [here](https://github.com/pulibrary/scripting-affinity-group).
